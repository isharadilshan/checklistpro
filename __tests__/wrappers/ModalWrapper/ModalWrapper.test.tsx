import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {View, Text} from 'react-native';

import ModalWrapper from '../../../src/components/wrappers/ModalWrapper';

configure({adapter: new Adapter()});

const props = {
  modalVisible: true,
  closeModal: jest.fn(),
  fullScreen: false,
  children: (
    <View>
      <Text>Test Text</Text>
    </View>
  ),
};

describe('when rendering', () => {
  const wrapper = shallow(<ModalWrapper {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
