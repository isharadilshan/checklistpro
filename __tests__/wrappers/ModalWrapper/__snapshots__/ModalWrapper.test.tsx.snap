// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={true}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  style={
    {
      "justifyContent": "center",
      "margin": 10,
    }
  }
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    [
      "down",
    ]
  }
  swipeThreshold={50}
  useNativeDriver={false}
>
  <withTheme(Surface)
    style={
      [
        {
          "borderRadius": 10,
          "elevation": 8,
        },
        {
          "backgroundColor": "#243B55",
          "flex": null,
        },
      ]
    }
  >
    <withTheme(IconButton)
      color="#BBC8D6"
      icon="close"
      onPress={[MockFunction]}
      size={25}
      style={
        {
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
    <View>
      <Text>
        Test Text
      </Text>
    </View>
  </withTheme(Surface)>
</ReactNativeModal>
`;
